arithmetization/src/main/java/net/consensys/linea/zktracer/module/modexpdata/ModexpDataOperation.java

```diff
@@ -0,0 +1,115 @@
+@Accessors(fluent = true)
+public class ModexpDataOperation extends ModuleOperation {
+  private static final int BEMR_LINE_COUNT = 32 * 4;
+  private static final int COMPONENT_SIZE = 512;
+
+  private final int hubStamp;
+
+  @EqualsAndHashCode.Include private final Bytes base;
+  @EqualsAndHashCode.Include private final Bytes exp;
+  @EqualsAndHashCode.Include private final Bytes mod;
+  @EqualsAndHashCode.Include private Bytes result;
+
+  private Bytes prevHubStamp = Bytes.EMPTY;
+
+  @Override
+  protected int computeLineCount() {
+    return BEMR_LINE_COUNT;
+  }
+
+  public void computeResult() {
+    BigInteger baseBigInt = base.toUnsignedBigInteger();
+    BigInteger expBigInt = exp.toUnsignedBigInteger();
+    BigInteger modBigInt = mod.toUnsignedBigInteger();
+
+    result = bigIntegerToBytes(baseBigInt.modPow(expBigInt, modBigInt));
+  }
+
+  void trace(Trace trace, int stamp) {
+    computeResult();
+
+    final UnsignedByte stampBytes = UnsignedByte.of(stamp);
+    Bytes currentHubStamp = Bytes.ofUnsignedInt(this.hubStamp + 1);
+
+    final Bytes basePadded = leftPadTo(base, COMPONENT_SIZE);
+    final Bytes expPadded = leftPadTo(exp, COMPONENT_SIZE);
+    final Bytes modPadded = leftPadTo(mod, COMPONENT_SIZE);
+    final Bytes resultPadded = leftPadTo(result, COMPONENT_SIZE);
+    final Bytes bemrLimb = Bytes.concatenate(basePadded, expPadded, modPadded, resultPadded);
+
+    BigInteger prevHubStampBigInt = prevHubStamp.toUnsignedBigInteger();
+    BigInteger hubStampBigInt = currentHubStamp.toUnsignedBigInteger();
+    BigInteger hubStampDiff = hubStampBigInt.subtract(prevHubStampBigInt).subtract(BigInteger.ONE);
+
+    Preconditions.checkArgument(
+        hubStampDiff.compareTo(BigInteger.valueOf(256 ^ 6)) < 0,
+        "Hub stamp difference should never exceed 256 ^ 6");
+
+    UnsignedByte[] hubStampDiffBytes =
+        bytesToUnsignedBytes(
+            rightPadTo(leftPadTo(bigIntegerToBytes(hubStampDiff), 6), 128).toArray());
+
+    for (int bemrIndex = 1; bemrIndex <= 4; bemrIndex++) {
+      UnsignedByte phase =
+          UnsignedByte.of(
+              switch (bemrIndex) {
+                case 1 -> Trace.PHASE_BASE;
+                case 2 -> Trace.PHASE_EXPONENT;
+                case 3 -> Trace.PHASE_MODULUS;
+                case 4 -> Trace.PHASE_RESULT;
+                default -> throw new IllegalStateException("Unexpected value: " + bemrIndex);
+              });
+
+      for (int index = 0; index < 32; index++) {
+        int counter = 32 * (bemrIndex - 1) + index;
+
+        trace
+            .ct(UnsignedByte.of(counter))
+            .bemr(UnsignedByte.of(bemrIndex))
+            .phase(phase)
+            .bytes(hubStampDiffBytes[counter])
+            .limb(bemrLimb.slice(16 * counter, 16))
+            .index(UnsignedByte.of(index))
+            .resultDataContext(currentHubStamp)
+            .stamp(stampBytes)
+            .validateRow();
+      }
+    }
+
+    prevHubStamp = currentHubStamp;
```

Comment:
ho the next modexOperatioData will be aware of the last hubStamp of the previous modexOperatioData ? Shouldn't it be given to ModexData ?

Reply:
Not exactly. `prevHubStamp` is an internal field that is only visible in the current `ModexpDataOperation` and does not leak in the next one.

Reply:
so how the second modexpOp will be aware of the first hubStamp ?